### PR TITLE
chore(vulnerability): Address high vulnerabilities

### PR DESCRIPTION
**Pull request by pinakipb2**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved randomness and security in test data generation by replacing `Math.random()` with `crypto.getRandomValues()` in various Cypress test files.
  - Enhanced error logging for JSON parsing and authorization events across multiple controllers for better error tracking and clarity.
  - Updated regex handling in schema translation utility for more robust field path segment extraction.
  - Simplified encryption algorithms in several components for improved security and performance.
  - Refined SSO callback logging to include protocol common name for better debugging insights.
  
- **Improvements**
  - Added validation for repository URL hosts in Git configurations to ensure only allowed hosts are processed.
  - Improved error message consistency and clarity in authentication services.

### CODE REVIEW

**Comment by jjoyce0510** [comment]
Thanks for the contribution! We are going to be reviewing soon! Can you please update the branch?

File: metadata-service/schema-registry-servlet/src/main/java/io/datahubproject/openapi/schema/registry/SchemaRegistryController.java [hunk 1]
```diff
@@ -299,7 +299,11 @@ public ResponseEntity<RegisterSchemaResponse> register(
             })
         .orElseGet(
             () -> {
-              log.error("Couldn't find topic with name {}.", topicName);
+              if (topicName.matches("^[a-zA-Z0-9._-]+$")) {
```
Comment (darnaut) [inline metadata-service/schema-registry-servlet/src/main/java/io/datahubproject/openapi/schema/registry/SchemaRegistryController.java:302]: Could you explain what is the security issue with printing the topic name?

File: metadata-service/factories/src/main/java/com/linkedin/metadata/boot/steps/IngestDataPlatformInstancesStep.java [hunk 2]
```diff
@@ -61,7 +61,7 @@ public void execute(@Nonnull OperationContext systemOperationContext) throws Exc
     long numEntities = _migrationsDao.countEntities();
     int start = 0;
 
-    while (start < numEntities) {
+    while (start < (int) numEntities) {
```
Comment (darnaut) [inline metadata-service/factories/src/main/java/com/linkedin/metadata/boot/steps/IngestDataPlatformInstancesStep.java:64]: Shouldn't we change the `start` type to long instead and fix all the relevant issues?

**Review by darnaut — CHANGES_REQUESTED** [review]

File: metadata-service/factories/src/main/java/com/linkedin/metadata/boot/steps/IngestDataPlatformInstancesStep.java (hunk 2 above)
Comment (pinakipb2) [reply metadata-service/factories/src/main/java/com/linkedin/metadata/boot/steps/IngestDataPlatformInstancesStep.java:64]: `start` is used in listAllUrns below, which expects int

File: metadata-service/schema-registry-servlet/src/main/java/io/datahubproject/openapi/schema/registry/SchemaRegistryController.java (hunk 1 above)
Comment (pinakipb2) [reply metadata-service/schema-registry-servlet/src/main/java/io/datahubproject/openapi/schema/registry/SchemaRegistryController.java:302]: Not exactly a security issue, but codeql was giving `log injection` vulnerability.
Similiar with other log changes as well.

**Comment by pinakipb2** [comment]
> Thanks for the contribution! We are going to be reviewing soon! Can you please update the branch?

Sure, have updated the branch.

File: metadata-service/schema-registry-servlet/src/main/java/io/datahubproject/openapi/schema/registry/SchemaRegistryController.java (hunk 1 above)
Comment (rtekal) [reply metadata-service/schema-registry-servlet/src/main/java/io/datahubproject/openapi/schema/registry/SchemaRegistryController.java:310]: topicName needs to be sanitized before logging.  If it cannot be sanitized, we are not logging it.

File: metadata-service/factories/src/main/java/com/linkedin/metadata/boot/steps/IngestDataPlatformInstancesStep.java (hunk 2 above)
Comment (darnaut) [reply metadata-service/factories/src/main/java/com/linkedin/metadata/boot/steps/IngestDataPlatformInstancesStep.java:64]: And we can't fix `listAllUrns`?

File: metadata-service/schema-registry-servlet/src/main/java/io/datahubproject/openapi/schema/registry/SchemaRegistryController.java (hunk 1 above)
Comment (darnaut) [reply metadata-service/schema-registry-servlet/src/main/java/io/datahubproject/openapi/schema/registry/SchemaRegistryController.java:310]: why does it need to be sanitized? this makes no sense at all

File: metadata-service/auth-servlet-impl/src/main/java/com/datahub/auth/authentication/AuthServiceController.java [hunk 3]
```diff
@@ -193,7 +193,7 @@ CompletableFuture<ResponseEntity<String>> signUp(final HttpEntity<String> httpEn
     try {
       bodyJson = mapper.readTree(jsonStr);
     } catch (JsonProcessingException e) {
-      log.debug("Failed to parse json while attempting to create native user", e);
+      log.error("Failed to parse json while attempting to create native user ", e);
```
Comment (darnaut) [inline metadata-service/auth-servlet-impl/src/main/java/com/datahub/auth/authentication/AuthServiceController.java:196]: why is this adding an space character at the end? it feels like this is some bad automation

File: metadata-service/auth-servlet-impl/src/main/java/com/datahub/auth/authentication/AuthServiceController.java [hunk 4]
```diff
@@ -238,7 +238,7 @@ CompletableFuture<ResponseEntity<String>> signUp(final HttpEntity<String> httpEn
           try {
             Urn inviteTokenUrn = _inviteTokenService.getInviteTokenUrn(inviteTokenString);
             if (!_inviteTokenService.isInviteTokenValid(systemOperationContext, inviteTokenUrn)) {
-              log.error("Invalid invite token {}", inviteTokenString);
+              log.error("Invalid invite token !");
```
Comment (darnaut) [inline metadata-service/auth-servlet-impl/src/main/java/com/datahub/auth/authentication/AuthServiceController.java:241]: what's the vulnerability here? why are we adding a ` !` at the end?

File: metadata-service/auth-servlet-impl/src/main/java/com/datahub/auth/authentication/AuthServiceController.java [hunk 5]
```diff
@@ -282,7 +282,7 @@ CompletableFuture<ResponseEntity<String>> resetNativeUserCredentials(
     try {
       bodyJson = mapper.readTree(jsonStr);
     } catch (JsonProcessingException e) {
-      log.debug("Failed to parse json while attempting to create native user", e);
+      log.error("Failed to parse json while attempting to create native user ", e);
```
Comment (darnaut) [inline metadata-service/auth-servlet-impl/src/main/java/com/datahub/auth/authentication/AuthServiceController.java:285]: why is this adding a space to the end of the string?

File: metadata-service/graphql-servlet-impl/src/main/java/com/datahub/graphql/GraphQLController.java [hunk 6]
```diff
@@ -75,7 +75,7 @@ CompletableFuture<ResponseEntity<String>> postGraphQL(
     try {
       bodyJson = mapper.readTree(jsonStr);
     } catch (JsonProcessingException e) {
-      log.error("Failed to parse json {}", jsonStr);
+      log.error("Failed to parse json ", e);
```
Comment (darnaut) [inline metadata-service/graphql-servlet-impl/src/main/java/com/datahub/graphql/GraphQLController.java:78]: what's the vulnerability here?

File: metadata-service/services/src/main/java/com/linkedin/metadata/service/RollbackService.java [hunk 7]
```diff
@@ -173,7 +173,7 @@ public RollbackResponse rollbackIngestion(
     // Rollback timeseries aspects
     DeleteAspectValuesResult timeseriesRollbackResult =
         timeseriesAspectService.rollbackTimeseriesAspects(opContext, runId);
-    rowsDeletedFromEntityDeletion += timeseriesRollbackResult.getNumDocsDeleted();
+    rowsDeletedFromEntityDeletion += timeseriesRollbackResult.getNumDocsDeleted().intValue();
```
Comment (darnaut) [inline metadata-service/services/src/main/java/com/linkedin/metadata/service/RollbackService.java:176]: why is `rowsDeletedFromEntityDeletion` not being converted to long?

File: smoke-test/tests/cypress/cypress/e2e/mutations/dataset_ownership.js [hunk 8]
```diff
@@ -1,4 +1,4 @@
-const test_id = Math.floor(Math.random() * 100000);
+const test_id = crypto.getRandomValues(new Uint32Array(1))[0];
```
Comment (darnaut) [inline smoke-test/tests/cypress/cypress/e2e/mutations/dataset_ownership.js:1]: what's the vulnerability here? this is for tests!

File: metadata-service/schema-registry-servlet/src/main/java/io/datahubproject/openapi/schema/registry/SchemaRegistryController.java (hunk 1 above)
Comment (darnaut) [reply metadata-service/schema-registry-servlet/src/main/java/io/datahubproject/openapi/schema/registry/SchemaRegistryController.java:310]: folks - the goal of logging is to log what the user is inputting. it's in fact more risky to give a user input to a regex than it is to log it. please when making such changes explain exactly what the issue is.

Comment (darnaut) [reply metadata-service/schema-registry-servlet/src/main/java/io/datahubproject/openapi/schema/registry/SchemaRegistryController.java:310]: https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS

**Comment by pinakipb2** [comment]
Created sub PRs addressing and explaining the causes of the vulnerabilities.

PRs:
- https://github.com/datahub-project/datahub/pull/11091
- https://github.com/datahub-project/datahub/pull/11090
- https://github.com/datahub-project/datahub/pull/11089
- https://github.com/datahub-project/datahub/pull/11088
- https://github.com/datahub-project/datahub/pull/11061
- https://github.com/datahub-project/datahub/pull/11060
- https://github.com/datahub-project/datahub/pull/11059
- https://github.com/datahub-project/datahub/pull/11058


Closing PR.

cc @rtekal

**Pull request closed without merging**